platform/fabric/core/generic/chaincode/discovery.go

```diff
@@ -127,15 +102,51 @@ func (d *Discovery) Call() ([]driver.DiscoveredPeer, error) {
 		})
 	}
 
-	// cache response
+	return discoveredEndorsers, nil
+}
+
+func (d *Discovery) Response() (discovery.Response, error) {
```

Comment:
Not sure this method describes what is actually going on here, maybe needs to be a bit more descriptive